netty/src/main/java/io/grpc/transport/netty/NettyServerStream.java

```diff
@@ -90,7 +92,24 @@ protected void internalSendHeaders(Metadata.Headers headers) {
   @Override
   protected void sendFrame(WritableBuffer frame, boolean endOfStream, boolean flush) {
     ByteBuf bytebuf = ((NettyWritableBuffer) frame).bytebuf();
-    channel.write(new SendGrpcFrameCommand(this, bytebuf, endOfStream));
+    final int numBytes = bytebuf.readableBytes();
+    if (numBytes > 0) {
```

Comment:
I'd prefer not to have a `numBytes == 0` optimization. It shouldn't ever happen on server-side right now, so the else won't ever be tested/run in practice.